src/components/Project/ProjectBasics/ProjectForm/ProjectStatusSection.tsx

```diff
@@ -93,16 +94,22 @@ const ProjectStatusSection: FC<IProjectStatusSectionProps> = ({
               ) {
                 return t('validation.phaseTooEarly', { value: phase.label });
               }
-              fields.push(
-                ...fieldsIfEmpty([
-                  ...programmedRequirements,
-                  ...planningRequirements,
-                  'estConstructionStart',
-                  'estConstructionEnd',
-                  'personConstruction',
-                  'constructionPhaseDetail',
-                ]),
-              );
+
+              if (phaseToSubmit !== warrantyPeriodPhase && phaseToSubmit !== completedPhase) {
+                fields.push(
+                  ...fieldsIfEmpty([
+                  ...combinedRequirements,
+                    'constructionPhaseDetail'
+                  ]),
+                );
+              } else {
+                fields.push(
+                  ...fieldsIfEmpty([
+                  ...combinedRequirements,
+                  ]),
+                );
+              }
+           
```

Comment:
Cause a switch statement is used here, instead of the if else, maybe something like 

          switch (phaseToSubmit) {
            case programmedPhase:
              fields.push(...fieldsIfEmpty([...programmedRequirements]));
              break;
            case draftInitiationPhase:
            case draftApprovalPhase:
            case constructionPlanPhase:
            case constructionWaitPhase:
              fields.push(...fieldsIfEmpty([...programmedRequirements, ...planningRequirements]));
              break;
            case constructionPhase:
              fields.push(
                ...fieldsIfEmpty([
                ...combinedRequirements,
                  'constructionPhaseDetail'
                ]),
              );
              break;
            case warrantyPeriodPhase:
            case completedPhase:
              if (
                isBefore(getToday(), getValues('estConstructionEnd'))
              ) {
                return t('validation.phaseTooEarly', { value: phase.label });
              }
              fields.push(
                ...fieldsIfEmpty([
                ...combinedRequirements,
                ]),
              );
              break;
          }

would be more readable 🤔

Reply:
I refactored it 👍🏻